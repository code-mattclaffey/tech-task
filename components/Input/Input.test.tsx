import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Input, Label, InputWrapper } from '.';

afterEach(() => {
    cleanup();
});

describe('<Input />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <Input id="input" name="input" />
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot when type is changed', () => {
        const { container } = render(
            <Input id="input" name="input" type="number" />
        );

        expect(container).toMatchSnapshot();
    });
});

describe('<Label />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <Label htmlFor="input">
                Some Label Text
            </Label>
        );

        expect(container).toMatchSnapshot();
    });
});

describe('<InputWrapper />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <InputWrapper>
                <Input id="input" name="input" type="number" />
                <Label htmlFor="input">
                    Some Label Text
                </Label>
            </InputWrapper>
        );

        expect(container).toMatchSnapshot();
    });
});
