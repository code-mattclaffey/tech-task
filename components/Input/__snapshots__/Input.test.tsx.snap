// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<div>
  <input
    class="c-input"
    id="input"
    name="input"
    type="text"
  />
</div>
`;

exports[`<Input /> should match snapshot when type is changed 1`] = `
<div>
  <input
    class="c-input"
    id="input"
    name="input"
    type="number"
  />
</div>
`;

exports[`<InputWrapper /> should match snapshot 1`] = `
<div>
  <div
    class="c-input__container"
  >
    <input
      class="c-input"
      id="input"
      name="input"
      type="number"
    />
    <label
      class="c-input__container"
      for="input"
    >
      Some Label Text
    </label>
  </div>
</div>
`;

exports[`<Label /> should match snapshot 1`] = `
<div>
  <label
    class="c-input__container"
    for="input"
  >
    Some Label Text
  </label>
</div>
`;
